Cython/Build/IpythonMagic.py

```diff
@@ -52,15 +52,36 @@
 import sys
 import time
 import copy
-import distutils.log
+
+try:
+    import distutils.log as logging
+    from distutils.log import set_threshold
+    from distutils.core import Distribution, Extension
+    from distutils.command.build_ext import build_ext
+    import distutils.errors as _setuptools_errors
+except ImportError:
+    import logging
+
+    def set_threshold(level):
```

Comment:
Without the qualifying module prefix, we'd better give this an explicit name (also for the import).
```suggestion
    def set_logging_threshold(level):
```